httpx/_client.py

```diff
@@ -763,7 +763,7 @@ def _send_handling_auth(
                 next_request = auth_flow.send(response)
             except StopIteration:
                 return response
-            except BaseException as exc:
+            except Exception as exc:  # noqa: PIE786
```

Comment:
Should we stick to catching `BaseException` here and in the other change in this file?

Reply:
I was thinking about that.

What the benefits of `BaseException`?

Catching `BaseException` there, we also handle `KeyboardInterrupt`, `GeneratorExit` and `SystemExit` and close the response in these cases. I don't think it's a proper behaviour (but may be I miss something)

Reply:
I'm okay with either, but since it would be a behavioral let's try and submit that as a separate PR, so that this only focuses on silencing any PIE786 errors?

Reply:
Sure, wait a second